src/atldld/sync.py

```diff
@@ -519,6 +520,172 @@ def corners_sagittal(r, dataset_id):
+        Tuple[int, float, np.ndarray, DisplacementField, np.ndarray],
+    ],
+    None,
+    None,
+]:
+    """Download entire dataset.
+
+    This function performs the following steps:
+
+    1. Get metadata for the entire dataset (e.g. `affine_3d`)
+    2. Get metadata for all images inside of the dataset (e.g. `affine_2d`)
+    3. For each image in the dataset do the following
+
+        a. Query the API to get the `p, i, r` coordinates of the `detection_xy`.
+        b. One of the `p, i, r` will become the `slice_coordinate`. For
+           coronal datasets it is the `p` and for sagittal ones it is the `r`.
+           In other words we assume that the slice is parallel to
+           one of the axes.
+        c. Use `get_transform_parallel` to get a full mapping between the
+           reference space and the image.
+        d. Download the image (+ potentially the expression image)
+        e. Yield result (order derived from section numbers - highest first)
+
+    Parameters
+    ----------
+    dataset_id
+        Id of the section dataset. Used to determine the 3D matrix.
+    ds_r
+        Downsampling factor of the reference
+        space. If set to 1 no downsampling takes place. The reference
+        space shape will be divided by `ds_r`.
+    detection_xy
+        Represents the x and y coordinate in the image that will be
+        used for determining the slice number in the reference space.
+        `p` for coronal slices, `r` for sagittal slices.
+    include_expression
+        If True then the generator returns 5 objects
+        where the last one is the expression image.
+    ds_i
+        The downloaded image will have both the height and the width
+        downsampled by `2 ** ds_i`.
+
+    Returns
+    -------
+    res_dict : generator
+        Generator yielding consecutive four tuples of
+        (image_id, constant_ref_coordinate, img, df).
+        The `constant_ref_coordinate` is the dimension in the given axis in microns.
+        The `img` is the raw gene expression image with dtype `uint8`.
+        The `df` is the displacement field of shape (8000 // `ds_r`, 11400 // `ds_r`).
+        Note that the sorting. If `include_expression=True` then last returned image
+        is the processed expression image.
+        That is the generator yield (image_id, p, img, df, img_expr).
+    """
+    metadata_2d = get_2d_bulk(
+        dataset_id,
+        ref2inp=True,
+    )
+    affine_3d = get_3d(
+        dataset_id,
+        ref2inp=True,
+        return_meta=False,
+    )
+    axis = CommonQueries.get_axis(dataset_id)
+
+    all_image_ids = sorted(
+        metadata_2d.keys(),
+        key=lambda x: -int(metadata_2d[x][1]),
+    )
+
+    for image_id in all_image_ids:
+        p, i, r = xy_to_pir_API_single(*detection_xy, image_id=image_id)
+
+        slice_ref_coordinate = p if axis == "coronal" else r
+        affine_2d = [v[0] for k, v in metadata_2d.items() if k == image_id][0]
```

Comment:
Make this more readable / transparent.

Reply:
Fixed in 88dfe0115d00f212e1e225e5af52505028f3c9d7. I hope it is more readable than before:)